### ISSUE TITLE
Bring back session.upload event

### ISSUE THREAD

**Issue by klizhentas** [gravitational/teleport#5088]
### Feature Request

Some teams would like to track `session.upload` event that is gone with stream refactoring. Emit the event at the end of the `stream` upload in async mode.

**Comment by russjones** [comment]
@quinqu 

Take a look at branch/4.3 you will be restoring the `session.upload` event from that release of Teleport.

For an example of how to create new events, take a look at the following PR and the AppSessionStart event, mirror that. Once you have the event, call `EmitAuditEvent`.

https://github.com/gravitational/teleport/pull/4582/files

Make sure `~/.aws/credentials` exists, then update `teleport.yaml` to have the following:

```yaml
  storage:
    type: "dynamodb"
    table_name: "jane-backend"
    audit_events_uri:  ["dynamodb://jane-audit-events"]
    audit_sessions_uri: "s3://jane-sessions"
```

Make sure you test success and failure modes.

Start with S3, then expand out to disk based, and IBM COS as well. Ideally you want your `session.upload` event hooked at a common point for all different upload types. That way you don't have to implement the `session.upload` event for three or four different providers.

**Comment by robertogiordani** [comment]
@quinqu when the PR will be included in the next 4.4.x version?

**Comment by quinqu** [comment]
@robertogiordani I am waiting for approval to merge and then will back port. Hopefully by early next week it will be available in 4.4.7.

**Comment by robertogiordani** [comment]
Hi Jane Quin,
does the 4.4.8 version include this fix?

Regards,
    Roberto.
________________________________________________________________________________

Roberto Giordani
Infrastructure Technical Security Lead 
IBM Rome Software Lab 
IBM Public Cloud
phone +39.348.4327340
E-mail roberto.giordani@it.ibm.com
WebMeeting https://ibm.webex.com/join/roberto.giordani
IBM Italia S.p.A. Via Luigi Stipa, 150 - 00148 Rome
